From ff249d6efb67e3f85e866b3b223b986d99f5614a Mon Sep 17 00:00:00 2001
From: "sukil.hong" <sukil.hong@lge.com>
Date: Tue, 18 Oct 2022 12:04:36 +0900
Subject: [PATCH] Fix compilation with lttng-ust >= 2.13

:Release Notes:
Fix compilation with lttng-ust >= 2.13

:Detailed Notes:
1. build error
|  error: #include expects "FILENAME" or <FILENAME>
|  1140 | #include LTTNG_UST_TRACEPOINT_INCLUDE
|       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~

long deprecated TRACEPOINT_INCLUDE_FILE macro was
finally removed in 2.13.5, see:
https://github.com/lttng/lttng-ust/commit/41858e2b6e8
https://github.com/lttng/lttng-ust/commit/d2966b4b0b2

2. hang issue
When webapp is executed, it forks the zygote process
and becoming a renderer process.
And it causes a hang when initialize lttng in renderer process.

:Testing Performed:
Build Test

:QA Notes:
N/A

:Issues Addressed:
[WRO-12611] LTTng tracing is not working

Change-Id: I281838e6810e05fda6e6af552d0a776a36b9dc2e
---
 files/launch/systemd/webapp-mgr.sh.in            | 5 ++++-
 src/core/trace/pmtrace_webappmanager3_provider.h | 6 +++---
 2 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/files/launch/systemd/webapp-mgr.sh.in b/files/launch/systemd/webapp-mgr.sh.in
index 9eddc5c..58cddc2 100644
--- a/files/launch/systemd/webapp-mgr.sh.in
+++ b/files/launch/systemd/webapp-mgr.sh.in
@@ -1,7 +1,7 @@
 #!/bin/sh
 # @@@LICENSE
 #
-# Copyright (c) 2017-2019 LG Electronics, Inc.
+# Copyright (c) 2017-2022 LG Electronics, Inc.
 #
 # Confidential computer software. Valid license from LG required for
 # possession, use or copying. Consistent with FAR 12.211 and 12.212,
@@ -31,6 +31,9 @@ export WAM_ERROR_PAGE="file:///usr/share/localization/wam/loaderror.html"
 # suspending javascript execution delay for page visibility
 export WAM_SUSPEND_DELAY_IN_MS=250
 
+# workaround to skip lttng 2.13 related issues
+export LTTNG_UST_WITHOUT_BADDR_STATEDUMP=1
+
 if [ -e "etc/wam/make_shm.sh" ] ; then
     /etc/wam/make_shm.sh
 fi
diff --git a/src/core/trace/pmtrace_webappmanager3_provider.h b/src/core/trace/pmtrace_webappmanager3_provider.h
index 35aa485..4b19b02 100644
--- a/src/core/trace/pmtrace_webappmanager3_provider.h
+++ b/src/core/trace/pmtrace_webappmanager3_provider.h
@@ -1,4 +1,4 @@
-// Copyright (c) 2013-2018 LG Electronics, Inc.
+// Copyright (c) 2013-2022 LG Electronics, Inc.
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
@@ -17,8 +17,8 @@
 #undef TRACEPOINT_PROVIDER
 #define TRACEPOINT_PROVIDER pmtrace_webappmanager3
 
-#undef TRACEPOINT_INCLUDE_FILE
-#define TRACEPOINT_INCLUDE_FILE pmtrace_webappmanager3_provider.h
+#undef TRACEPOINT_INCLUDE
+#define TRACEPOINT_INCLUDE "pmtrace_webappmanager3_provider.h"
 
 #ifdef __cplusplus
 extern "C" {
-- 
2.37.1

